crates/rustc_codegen_spirv/src/abi.rs

```diff
@@ -168,14 +169,110 @@ pub(crate) fn provide(providers: &mut Providers) {
         };
 
         if hide_niche {
-            layout = tcx.mk_layout(LayoutS {
+            layout = tcx.mk_layout(LayoutData {
                 largest_niche: None,
                 ..clone_layout(layout.0.0)
             });
         }
 
         Ok(TyAndLayout { ty, layout })
     };
+
+    // HACK(eddyb) work around https://github.com/rust-lang/rust/pull/129403
+    // banning "struct-style" `#[repr(simd)]` (in favor of "array-newtype-style"),
+    // by simply bypassing "type definition WF checks" for affected types, which:
+    // - can only really be sound for types with trivial field types, that are
+    //   either completely non-generic (covering most `#[repr(simd)]` `struct`s),
+    //   or *at most* one generic type parameter with no bounds/where clause
+    // - relies on upstream `layout_of` not having had the non-array logic removed
+    //
+    // FIXME(eddyb) remove this once migrating beyond `#[repr(simd)]` becomes
```

Comment:
Prob want a task for this too.

Reply:
- https://github.com/Rust-GPU/rust-gpu/issues/182